js/compiler/job/Job.js

```diff
@@ -18,6 +18,9 @@ export class Job {
     #stdout = '';
 
     #stopped = false;
+    
+    /** @type {number|null} */
+    #exitCode = null;
```

Comment:
I'd prefer going with something like a union type for something like this, where if we need the exit code it could be something more clear like:

```ts
declare type JobStatus = 
    { status: 'running' } |
    { status: 'stopped', stoppedByUser: boolean, exitCode: number };
```

And then `#stopped` and `#exitCode` can be wrapped into one while providing some nicer intellisense, and with no sentinel `-1` value :P